commit c5d1edd81633fea56ab4e4dfe15fdc1de45879ba
Author: Yang Bo <yangbo@deepin.com>
Date:   Wed Feb 8 10:36:31 2017 +0800

    elantech: try synaptics style sliced command if fast command failed.
    
    Signed-off-by: Yang Bo <yangbo@deepin.com>

diff --git a/drivers/input/mouse/elantech.c b/drivers/input/mouse/elantech.c
index db7d1d666..01a5ae2e8 100644
--- a/drivers/input/mouse/elantech.c
+++ b/drivers/input/mouse/elantech.c
@@ -55,8 +55,13 @@ static int elantech_send_cmd(struct psmouse *psmouse, unsigned char c,
 	if (ps2_command(ps2dev, NULL, ETP_PS2_CUSTOM_COMMAND) ||
 	    ps2_command(ps2dev, NULL, c) ||
 	    ps2_command(ps2dev, param, PSMOUSE_CMD_GETINFO)) {
-		psmouse_err(psmouse, "%s query 0x%02x failed.\n", __func__, c);
-		return -1;
+			/* try synaptics style sliced command if
+			 * fast command failed.
+			 * */
+			if (synaptics_send_cmd(psmouse, c, param)) {
+				psmouse_err(psmouse, "%s query 0x%02x failed.\n", __func__, c);
+				return -1;
+			}
 	}
 
 	return 0;
